woocommerce/woocommerce-android
# [Order Creation] Display shipping fees on the form + edit flow

Closes: #5698 


### Description
This PR has two main changes:
1. Display the shipping fee values in the payment section.
2. Implement the edit flow, including shipping fee removal.

The removal part depends on the FluxC [PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2290)

### Testing instructions
1. Make sure the feature flag ORDER_CREATION_M2 is enabled.
2. Open order creation form.
3. Add a product.
5. Click on "Add shipping".
6. Enter an amount and a name then add a shipping fee.
7. Notice that the shipping fee is displayed on the payment section, and that the button was changed to "Shipping".
8. Click on the "Shipping" button.
9. Confirm that the current shipping fee data is displayed, and the presence of "Remove shipping" button.
10. Make some edits to the shipping data, and click on "Done".
11. Confirm the payment section is updated, and an order draft sync is triggered.
12. Click on the "Shipping" button.
13. Click on "Remove shipping" button.
14. Confirm the payment section has been updated, with a draft sync trigger.

### Images/gif
<img width="360" alt="Screen Shot 2022-02-21 at 14 59 26" src="https://user-images.githubusercontent.com/1657201/154969396-c5247984-cd02-40ca-9c28-ce7a66399e6f.png"> <img width="360" alt="Screen Shot 2022-02-21 at 14 59 32" src="https://user-images.githubusercontent.com/1657201/154969407-7fd25542-3a6e-4b21-849b-4af4cd8e2449.png">
